### DISCUSSION TITLE
Could not find module entry with specifier "garden/app

### DISCUSSION THREAD

**Discussion by mirhossain8248** [lukethacoder/lwc-garden#6] [Q&A] [answered]
`'0': {
    description: {
      category: 'lwrUnresolvable/module',
      advice: [],
      message: '404: Could not find module entry with specifier "garden/app"'
    }
  },
  length: 1
}
[error] DiagnosticsError: 404: Could not find module entry with specifier "garden/app"`

here's the config file:
```

{
    "lwc": {
      "modules": [
        {
          "dir": "./.garden/components",
          "namespace": "garden"
        },
        {
          "dir": "./force-app/main/default/lwc",
          "namespace": "c"
        }
      ]
    },
    "routes": [
      {
        "id": "app",
        "path": "/",
        "rootComponent": "garden/app",
        "layoutTemplate": "./.garden/layouts/index.html",
        "bootstrap": {
          "syntheticShadow": true
        }
      }
    ]
  }
  
```
  
  
<img width="959" alt="Screenshot 2024-04-12 153658" src="https://github.com/lukethacoder/lwc-garden/assets/37155763/46fedae8-4f6d-4679-ab55-3abff1984d12">

Getting this error after trying to open the local host link, double checked and I do have the app file in the garden directory. Anything else I should be looking out for?

**Comment by lukethacoder** [comment]
Have you got an example repo to share? 

What operating system and lwc-garden/core version are you using?

**Reply by mirhossain8248** [reply]
Here's the example repo:
https://github.com/mirhossain8248/LWC-Garden-Test

OS: Windows 11
lwc-garden version: not sure I ran through the documentation, I assume I have the most recent one as of Friday (03/12/2024)

**Comment by lukethacoder** [comment, marked as answer]
Looks like you're missing a few packages and config. I'll update the documentation site shortly to make things a bit more clear.

Here is a live working example if you need something to compare against:
https://stackblitz.com/edit/lwc-garden-minimal-config?file=lwr.config.json


Looks like you are missing 2 key things:

#### 1. missing [`@lukethacoder/lwc-module-resolver`](https://github.com/lukethacoder/lwc-module-resolver) and [`@lukethacoder/lwrjs-lwc-module-provider`](https://github.com/lukethacoder/lwrjs-lwc-module-provider) packages
These handle the bespoke LWC `module` config. Can learn more about it [here](https://github.com/lukethacoder/lwc-module-resolver).

#### 2. Missing the `moduleProviders` config in `lwr.config.json`

```json
{
  "lwc": {
    "modules": [
      {
        "dir": "./.garden/components",
        "namespace": "garden"
      },
      {
        "dir": "./force-app/main/default/lwc",
        "namespace": "c"
      }
    ]
  },
  "moduleProviders": [
    "@lwrjs/label-module-provider",
    "@lwrjs/app-service/moduleProvider",
    "@lukethacoder/lwrjs-lwc-module-provider",
    "@lwrjs/npm-module-provider"
  ],
  "routes": [
    {
      "id": "app",
      "path": "/",
      "rootComponent": "garden/app",
      "layoutTemplate": "./.garden/layouts/index.html",
      "bootstrap": {
        "syntheticShadow": true
      }
    }
  ]
}
```

---

I can see you're trying to use `lightning` too. The above fixes will fix the core local dev server, but you will need a bit more config to get `lightning-base-components` working.

Checkout the [documentation](https://lwc.garden/guides/lightning-base-components/) and the [live example](https://stackblitz.com/edit/lwc-garden-lightning-base-components?file=package.json) to get them working.

**Reply by mirhossain8248** [reply]
Thank you! I'll try these out when I get some time!
